arches_references/management/commands/controlled_lists.py

```diff
@@ -169,15 +171,21 @@ def migrate_collections_to_controlled_lists(
             result = cursor.fetchone()
             self.stdout.write(result[0])
 
-    def migrate_concept_nodes_to_reference_datatype(self, graph_id):
+    def migrate_concept_nodes_to_reference_datatype(self, graph):
         try:
-            source_graph = GraphModel.objects.get(pk=graph_id)
+            try:
+                UUID(str(graph))
+                query = models.Q(graphid=graph)
+            except (ValueError, TypeError):
+                query = models.Q(slug=graph, source_identifier__isnull=True)
+            source_graph = GraphModel.objects.get(query)
+            graph_id = source_graph.graphid
         except (GraphModel.DoesNotExist, ValidationError) as e:
             raise CommandError(e)
```

Comment:
The double-nested try's and the double-cast from str to UUID feels a little too complex to me.

I would suggest a solution along the lines of:
- use isinstance(... to find out if you already have a UUID
- catch only ValueError from UUID()
- don't catch ValidationError

In general I try to have only statements in a `try` that have a chance of failing. There are several statements in here that would never fail.